test/common/http/http2/http2_frame.cc

```diff
@@ -153,16 +155,53 @@ Http2Frame Http2Frame::makePriorityFrame(uint32_t stream_index, uint32_t depende
   Http2Frame frame;
   frame.buildHeader(Type::Priority, kPriorityPayloadSize, 0, makeRequestStreamId(stream_index));
   uint32_t dependent_net = makeRequestStreamId(dependent_index);
+  ASSERT(frame.data_.capacity() >= HeaderSize + sizeof(uint32_t));
   memcpy(&frame.data_[HeaderSize], reinterpret_cast<void*>(&dependent_net), sizeof(uint32_t));
   return frame;
 }
 
+Http2Frame Http2Frame::makeEmptyPushPromiseFrame(uint32_t stream_index,
+                                                 uint32_t promised_stream_index,
+                                                 HeadersFlags flags) {
+  static constexpr size_t kEmptyPushPromisePayloadSize = 4;
+  Http2Frame frame;
+  frame.buildHeader(Type::PushPromise, kEmptyPushPromisePayloadSize, static_cast<uint8_t>(flags),
+                    makeRequestStreamId(stream_index));
+  uint32_t promised_stream_id = makeRequestStreamId(promised_stream_index);
```

Comment:
Nit: best to `const` qualify everything that can be, even locals.

Reply:
Done